db/src/db.rs

```diff
@@ -515,12 +515,19 @@ pub trait MentatStoring {
     fn insert_non_fts_searches<'a>(&self, entities: &'a [ReducedEntity], search_type: SearchType) -> Result<()>;
     fn insert_fts_searches<'a>(&self, entities: &'a [ReducedEntity], search_type: SearchType) -> Result<()>;
 
-    /// Finalize the underlying storage layer after a Mentat transaction.
-    ///
+    /// Prepare the underlyin storage layer for finalization after a Mentat transaction.
+    /// 
     /// Use this to finalize temporary tables, complete indices, revert pragmas, etc, after the
     /// final `insert_non_fts_searches` invocation.
+    fn materialize_transaction(&self, tx_id: Entid) -> Result<()>;
```

Comment:
I got tripped up earlier this week, so let's call this `materialize_mentat_transaction`.